From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Otto=20Pfl=C3=BCger?= <otto.pflueger@abscue.de>
Date: Mon, 27 Apr 2026 20:43:21 +0200
Subject: [PATCH 013/225] spi: sprd: don't fail on unsupported delay unit when
 delay is 0
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

If the word delay is set to 0, which it is for most transfers, it does
not matter which unit is used because there will be no delay, so no
special handling is needed. Skip the unit check for such transfers to
avoid needlessly failing with -EINVAL.

Signed-off-by: Otto Pflüger <otto.pflueger@abscue.de>
---
 drivers/spi/spi-sprd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-sprd.c b/drivers/spi/spi-sprd.c
index acebf9c2e..676fcd69f 100644
--- a/drivers/spi/spi-sprd.c
+++ b/drivers/spi/spi-sprd.c
@@ -667,7 +667,7 @@ static int sprd_spi_init_hw(struct sprd_spi *ss, struct spi_transfer *t)
 	u16 word_delay, interval;
 	u32 val;
 
-	if (d->unit != SPI_DELAY_UNIT_SCK)
+	if (d->value != 0 && d->unit != SPI_DELAY_UNIT_SCK)
 		return -EINVAL;
 
 	val = readl_relaxed(ss->base + SPRD_SPI_CTL0);
